microprot/scripts/pdb_search.py

```diff
@@ -388,3 +391,156 @@ def report_uncovered_subsequences(hits, query, min_subseq_len=40):
 
     return [u for u in uncovered if (u['sequence'] != '') &
                                     (len(u['sequence']) >= min_subseq_len)]
+
+
+def get_q_id(hit):
+    """ Returns the query ID for a hit.
+
+    Parameters
+    ----------
+    A hit parsed from an HHsearch output file, i.e. dict with at least the key
+    'alignment' which is a dict by itself and comes at least with the key
+    'Q xxx' where xxx is some identifier. The value for this 'Q xxx' key is a
+    third dict which needs to contain the key 'sequence'.
+
+    Returns
+    -------
+    str : The query ID starting with 'Q '.
+    """
+    # find the right ID
+    id_ = [id_ for id_ in hit['alignment'].keys()
+           if id_.startswith('Q')
+           and id_ != 'Q Consensus'][0]
+    return id_
+
+
+def frag_size(hit):
+    """ Compute the fragment length of a hit.
+
+    Parameters
+    ----------
+    A hit parsed from an HHsearch output file, i.e. dict with at least the key
+    'alignment' which is a dict by itself and comes at least with the key
+    'Q xxx' where xxx is some identifier. The value for this 'Q xxx' key is a
+    third dict which needs to contain the key 'sequence'.
+
+    Returns
+    -------
+    The length of the un-gapped sequence for the given hit."""
+    # find the right ID
+    id_ = get_q_id(hit)
```

Comment:
I think that private variable names should have leading underscore instead of trailing underscore.